packages/graphile-build-pg/src/QueryBuilder.js

```diff
@@ -925,6 +950,7 @@ order by (row_number() over (partition by 1)) desc`; /* We don't need to factor
     this.lock("limit");
     this.lock("first");
     this.lock("last");
+    this.lock("distinctOn");
```

Comment:
I'm concerned that because `distinctOn` changes the order, it may break connections (specifically cursors) in some cases. This is one of the reasons that we lock orderBy when we do. E.g. if you order by `[USERNAME_ASC, NAME_DESC, ID_ASC]` and then `distinctOn: [NAME]` then the order would be changed to `NAME_DESC, USERNAME_ASC, ID_ASC` and pagination would work differently.

If you don't want to dig into this, maybe the best bet is to forbid combining distinctOn with connections - e.g. before setting distinctOn you lock cursorComparator/selectCursor/etc and then you enforce that they aren't used (otherwise you throw).

Reply:
I have added the appropriate locks as I don't have the time to dig into it. I hope I got the locks correct.

From the little research I did cursors should work correctly with `DISTINCT ON`